src/pages/guide/integration/data-fixtures-guide.md

```diff
@@ -0,0 +1,673 @@
+    $product2 = $fixtures->get('product2');
+}
+```
+
+**With fixture references:**
+
+```php
+/**
+ * @magentoDataFixture \Magento\Catalog\Test\Fixture\Product with:{"sku": "simple1", "price": 5.0} as:product1
+ * @magentoDataFixture \Magento\Catalog\Test\Fixture\Product with:{"sku": "simple2", "price": 10.0} as:product2
+ * @magentoDataFixture \Magento\Quote\Test\Fixture\GuestCart as:cart
+ * @magentoDataFixture \Magento\Quote\Test\Fixture\AddProductToCart with:{"cart": "$cart$", "product": "$product1$", "qty": 2}
+ * @magentoDataFixture \Magento\Quote\Test\Fixture\AddProductToCart with:{"cart": "$cart$", "product": "$product2$", "qty": 1}
+ */
+public function testCartWithProducts(): void
+{
+    $fixtures = DataFixtureStorageManager::getStorage();
+    $cart = $fixtures->get('cart');
+    // Cart now has 2 units of product1 and 1 unit of product2
+}
+```
+
+**Without parameters (uses defaults):**
+
+```php
+/**
+ * @magentoDataFixture \Magento\Catalog\Test\Fixture\Product as:product
+ */
+public function testProductWithDefaults(): void
+{
+    $fixtures = DataFixtureStorageManager::getStorage();
+    $product = $fixtures->get('product');
+    // Product created with default values (auto-generated unique SKU, etc.)
+}
+```
+
+#### Using legacy file-based fixtures (Deprecated)
+
+```php
+/**
+ * @magentoDataFixture Magento/Catalog/_files/product_simple.php
+ * @magentoDataFixture Magento/Customer/_files/customer.php
+ */
+public function testWithLegacyFixtures(): void
+{
+    // DEPRECATED - New legacy fixtures cannot be created
+    // Use parametrized data fixtures instead
+}
+```
+
+## Migration guide: Legacy to Modern
+
+If you have tests using legacy fixtures (file-based with `@magentoDataFixture`), you should migrate them to parametrized data fixtures. New legacy fixtures **cannot be created**.
+
+### Before (Legacy fixtures with DocBlock annotations)
+
+```php
+/**
+ * @magentoDataFixture Magento/Catalog/_files/product_simple.php
+ */
+public function testProductPrice(): void
+{
+    $product = $this->productRepository->get('simple'); // Fixed SKU
+    $this->assertEquals(10, $product->getPrice()); // Fixed price
+}
+```
+
+### After (Option 1: Parametrized data with Annotations)
+
+```php
+/**
+ * @magentoDataFixture \Magento\Catalog\Test\Fixture\Product with:{"sku": "test-product", "price": 10} as:product
+ */
+public function testProductPrice(): void
+{
+    $fixtures = DataFixtureStorageManager::getStorage();
+    $product = $fixtures->get('product');
+    $this->assertEquals(10, $product->getPrice());
+}
+```
+
+### After (Option 2: Parametrized data with PHP Attributes)
+
+```php
+#[
+    DataFixture(ProductFixture::class, ['sku' => 'test-product', 'price' => 10], 'product')
+]
+public function testProductPrice(): void
+{
+    $fixtures = DataFixtureStorageManager::getStorage();
+    $product = $fixtures->get('product');
+    $this->assertEquals(10, $product->getPrice());
+}
+```
+
+## Common fixture examples
```

Comment:
Please review all the examples and make sure the configurations are correct. I would recommend taking these examples straight from existing tests